pool/sender_pool.go

```diff
@@ -0,0 +1,75 @@
+package pool
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	qdb "github.com/questdb/go-questdb-client/v3"
+)
+
+type LineSenderPool struct {
```

Comment:
The pool should have `Close(ctx)` method that would close all remaining senders and turn the pool into `closed` mode so that it closes any released sender.